pkg/controller/instance/instance_controller.go

```diff
@@ -316,7 +307,54 @@ func instanceEventFilter(mgr manager.Manager) predicate.Funcs {
 	}
 }
 
-func createPlan(mgr manager.Manager, planName string, instance *kudov1alpha1.Instance) error {
+func createPlan(r *ReconcileInstance, planName string, instance *kudov1alpha1.Instance) error {
+	ctx := context.TODO()
+	gvk, _ := apiutil.GVKForObject(instance, r.scheme)
+	log.Printf("Creating PlanExecution of plan %s for instance %s", planName, instance.Name)
+	r.recorder.Event(instance, "Normal", "CreatePlanExecution", fmt.Sprintf("Creating \"%v\" plan execution", planName))
+
+	ref := corev1.ObjectReference{
+		Kind:      gvk.Kind,
+		Name:      instance.Name,
+		Namespace: instance.Namespace,
+	}
+
+	planExecution := kudov1alpha1.PlanExecution{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%v-%v-%v", instance.Name, planName, time.Now().Nanosecond()),
+			Namespace: instance.GetNamespace(),
+			// TODO: Should also add one for Operator in here as well.
+			Labels: map[string]string{
+				kudo.OperatorVersionAnnotation: instance.Spec.OperatorVersion.Name,
+				kudo.InstanceLabel:             instance.Name,
+			},
+		},
+		Spec: kudov1alpha1.PlanExecutionSpec{
+			Instance: ref,
+			PlanName: planName,
+		},
+	}
+
+	// Make this instance the owner of the PlanExecution
+	if err := controllerutil.SetControllerReference(instance, &planExecution, r.scheme); err != nil {
+		log.Printf("InstanceController: Error setting ControllerReference")
+		return err
+	}
+
+	if err := r.Create(ctx, &planExecution); err != nil {
+		log.Printf("InstanceController: Error creating planexecution \"%v\": %v", planExecution.Name, err)
+		r.recorder.Event(instance, "Warning", "CreatePlanExecution", fmt.Sprintf("Error creating planexecution \"%v\": %v", planExecution.Name, err))
+		return err
+	}
+	log.Printf("Created PlanExecution of plan %s for instance %s", planName, instance.Name)
+	r.recorder.Event(instance, "Normal", "PlanCreated", fmt.Sprintf("PlanExecution \"%v\" created", planExecution.Name))
+	return nil
+}
+
+//todo: REMOVE this... we do not want to create a plan outside of reconcile
+// most likely... any use of this function is wrong... all plans should be created in reconcile
+func createPlan2(mgr manager.Manager, planName string, instance *kudov1alpha1.Instance) error {
```

Comment:
`createPlanDeprecated`? I wasn't sure if `2` was the new or old one.

Reply:
I'll switch to `Old`